internal/utils/utils.go

```diff
@@ -0,0 +1,47 @@
+package utils
+
+import (
+	"sort"
+	"strings"
+)
+
+// FindLongestCommonPrefix finds the longest common path prefix of a set of paths. For example,
+// given the following:
+//
+//	github.com/owner/repo/cmd/foo
+//	github.com/owner/repo/cmd/bar
+//
+// The longest common prefix is: github.com/owner/repo/cmd/ (note the trailing slash is included).
+func FindLongestCommonPrefix(paths []string) string {
+	if len(paths) < 2 {
+		return ""
+	}
+	// Sort the paths to optimize comparison.
+	sort.Strings(paths)
+
+	first, last := paths[0], paths[len(paths)-1]
+	if first == last {
+		return first
+	}
```

Comment:
What happens if the path is something like

foo/bar/foo

Or something like that, my point is if the first last are equal.

I could happen when the project has a `replace ` in the go.mod file to rewrite long project URLs to something shorter

Reply:
It depends on what the other paths are. I added a test case just in case, but this is what I'd expect:

```go
{
	paths: []string{
		"foo/bar/foo",
		"foo/foo/foo",
	},
	want: "foo/",
},
```